### DISCUSSION TITLE
Properly Incorporating Your Updates To My Chirpy Blog

### DISCUSSION THREAD

**Discussion by bethanybeachbum** [cotes2020/jekyll-theme-chirpy#1634] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

1. I created a blog from your Chirpy-Starter.  It is successfully up and running.  
2. I have not done any customizations - I want to make easy to keep up with your updates.
3. As you come out with new updates, I want to incorporate them into my clone of Chirpy-Starter.  
4. Is the best place to find your latest version of Chirpy in the file: "jekyll-theme-chirpy.gemspec", line 5:  "spec.version       = "6.5.5" ?
5. Is the best place to find my version of Chirpy in the GEMfile --> line 5: gem "jekyll-theme-chirpy", "~> 6.5", ">= 6.5.5" ?

### Operations you have already tried

I believe the correct methodology for upgrading is:
From README.md in jekyll-theme-chirpy --> Documentation --> Wiki --> Upgrade Guide --> Upgrade From Starter --> execute code instructions

Is this all correct?

### Anything else?

my site is bethanybeachbum.github.io
Thank you for a great github theme that you excellently maintain.

**Comment by kungfux** [comment, marked as answer]
Yes, that's all correct.

I would just add that you can find the Chirpy version on the [Releases tab](https://github.com/cotes2020/jekyll-theme-chirpy/releases), which contains all the release notes.

Also, not every release requires you to manually update your version number. By default, every time your blog is built and deployed, it will use the latest available patch version. You have control over this process, and it depends on the dependency pattern you define. You can read more about [Ruby patterns](https://guides.rubygems.org/patterns/#pessimistic-version-constraint). So if your pattern is defined as `~> 6.5 >= 6.5.5`, then when version `6.5.6` is released, all you have to do is build and publish your blog, and it will use the latest version. However, when major version `7.0` is released, you will need to follow the upgrade guide you mentioned and make changes to `Gemfile` like `~> 7.0 >= 7.0.0`. This is a necessary step, as the major release will likely have breaking changes that you need to fix before you can publish your blog with it.

**Comment by zobayer1** [comment]
Is this process the same as when I created the site by cloning the repo instead of using the starter?

**Reply by kungfux** [reply]
No, it's different. You can find some insights in the [Upgrade Guide](https://github.com/cotes2020/jekyll-theme-chirpy/wiki/Upgrade-Guide).
